main.go

```diff
@@ -24,6 +24,7 @@ func (*handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	URL := getRequestURL(w, r)
 	if URL == nil { // invalid URL
+		serveLandingPage(w, r)
```

Comment:
Do not call the landing page here...
As `URL` can be `nil` in some other case also, in such cases the `Return` method errors are overwritten.
```suggestion
```